lib/libzstd/Makefile.am

```diff
@@ -0,0 +1,45 @@
+include $(top_srcdir)/config/Rules.am
+
+VPATH = \
+	$(top_srcdir)/contrib/zstd \
+	$(top_srcdir)/contrib/zstd/common \
+	$(top_srcdir)/contrib/zstd/compress \
+	$(top_srcdir)/contrib/zstd/decompress \
+	$(top_srcdir)/module/zstd
+
+DEFAULT_INCLUDES += \
+	-I$(top_srcdir)/contrib/zstd \
+	-I$(top_srcdir)/contrib/zstd/common \
+	-I$(top_srcdir)/contrib/zstd/compress \
+	-I$(top_srcdir)/contrib/zstd/decompress
+
+noinst_LTLIBRARIES = libzstd.la
+
+KERNEL_C = \
+	zstd_common.c \
+	fse_decompress.c \
+	entropy_common.c \
+	xxhash.c \
+	hist.c \
+	error_private.c \
+	zstd_compress.c \
+	zstd_compress_literals.c \
+	zstd_compress_sequences.c \
+	fse_compress.c \
+	huf_compress.c \
+	zstd_double_fast.c \
+	zstd_fast.c \
+	zstd_lazy.c \
+	zstd_ldm.c \
+	zstd_opt.c \
+	zstd_ddict.c \
+	zstd_decompress.c \
+	zstd_decompress_block.c \
+	huf_decompress.c \
+	zstd.c
+
+nodist_libzstd_la_SOURCES = $(KERNEL_C)
+
+# -fno-tree-vectorize gets set for gcc in zstd/common/compiler.h
+# Set it for other compilers, too.
+CFLAGS_zstd_decompress_block.o := -fno-tree-vectorize
```

Comment:
Compiler does not like := though:
`:= assignments are not portable`

Reply:
it's not the compiler but automake; the problem here is that automake does not "understand" :=, so it warns. However, automake will copy that line straight to the Makefile, which is the intended behaviour here

Reply:
(I leave that unresolved for a while. Maybe someone has a better idea how to solve that. I'm sure there is a cleaner way, that I just didn't find, yet.)